File: app/src/test/java/com/socotech/swa/net/ApiEndpointTest.java
package com.socotech.swa.net;

import com.socotech.swa.mvp.UnitTestSchedulerProvider;
import io.reactivex.observers.TestObserver;
import org.junit.Before;
import org.junit.Test;
import org.mockito.MockitoAnnotations;
import retrofit2.Response;

/**
 * Example local unit test, which will execute on the development machine (host).
 *
 * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
 */
public class ApiEndpointTest {

    private RandomUserApi randomUserApi;

    @Before
    public void onSetup() {
        MockitoAnnotations.initMocks(this);
        randomUserApi = new NetModule().api(new UnitTestSchedulerProvider());
    }

    @Test
    public void testGet() {
        TestObserver<Response<RandomUsers>> observable = randomUserApi.get("us", 1, "swa_challenge", 100).test();
        observable.assertNoErrors();
    }

}

File: app/src/test/java/com/example/net/ApiEndpointTest.kt
package com.example.net

import com.android.net.NetModule
import com.android.net.RandomUserApi
import com.example.mvp.UnitTestSchedulerProvider
import org.junit.Before
import org.junit.Test

/**
 * Example local unit test, which will execute on the development machine (host).
 *
 * @see [Testing documentation](http://d.android.com/tools/testing)
 */
class ApiEndpointTest {

    lateinit var randomUserApi: RandomUserApi

    @Before
    fun onSetup() {
        randomUserApi = NetModule().api(UnitTestSchedulerProvider())
    }

    @Test
    fun testGet() {
        val observable = randomUserApi.get("us", 1, "swa_challenge", 100).test()
        observable.assertNoErrors()
    }

}